cmd/pravega-operator/main.go

```diff
@@ -2,25 +2,43 @@ package main
 
 import (
 	"context"
+	"flag"
+	"fmt"
+	"os"
 	"runtime"
 	"time"
 
 	"github.com/operator-framework/operator-sdk/pkg/sdk"
 	sdkVersion "github.com/operator-framework/operator-sdk/version"
 	"github.com/pravega/pravega-operator/pkg/stub"
 	"github.com/pravega/pravega-operator/pkg/utils/k8sutil"
+	"github.com/pravega/pravega-operator/pkg/version"
 	"github.com/sirupsen/logrus"
 	_ "k8s.io/client-go/plugin/pkg/client/auth/gcp"
 )
 
-func printVersion() {
-	logrus.Infof("Go Version: %s", runtime.Version())
-	logrus.Infof("Go OS/Arch: %s/%s", runtime.GOOS, runtime.GOARCH)
-	logrus.Infof("operator-sdk Version: %v", sdkVersion.Version)
+var (
+	printVersion bool
+)
+
+func init() {
+	flag.BoolVar(&printVersion, "version", false, "Show version and quit")
+	flag.Parse()
 }
 
 func main() {
-	printVersion()
+	if printVersion {
+		fmt.Println("pravega-operator Version:", version.Version)
+		fmt.Println("Go Version:", runtime.Version())
+		fmt.Printf("Go OS/Arch: %s/%s\n", runtime.GOOS, runtime.GOARCH)
+		fmt.Printf("operator-sdk Version: %v", sdkVersion.Version)
+		os.Exit(0)
+	}
```

Comment:
@mcgG Yes, I agree with you that printing the version and exiting doesn't make sense in a k8 deployment. However, I think it can be useful for troubleshooting cases, e.g., when you don't know what's the version of a running operator process. Furthermore, this is just the first flag, but we can add more flags in the future to customize the behavior of the operator, e.g., watch all namespaces, metrics listen port, etc.

You can pass these options to the operator like this:

```
...
spec:
      containers:
        - name: pravega-operator
          image: maddisondavid/pravega-operator:latest
          ports:
          - containerPort: 60000
            name: metrics
          command:
          - pravega-operator
          - -flag1
          - -flag2
          imagePullPolicy: Always
...
```

or if you run the operator locally:

```
$ operator-sdk up local --operator-flags "-version"
```